### DISCUSSION TITLE
Adding 1-wire sensor to dashboard

### DISCUSSION THREAD

**Discussion by ejbones** [bareboat-necessities/lysmarine_gen#249] [Q&A]
Hey there, just installed BBN and it is looking great so far. I have limited experience with linux, and am trying to get temperatures via 1-wire devices. The manual states to use signalk.

I've already enables 1-wire on the pi and in the terminal I believe I can see it using "ls /sys/bus/w1/devices/". It returns an address at "bus master1".

What would be the framework needed to get a dashboard w/ multiple temperature DS18B20 1-wire sensors?

**Comment by mgrouch** [comment]
Here is one way using NodeRed to SignalK
https://pysselilivet.blogspot.com/2018/07/signalk-measuring-temperature.html


Another is just SignalK plugin
https://github.com/ewaldvangemert/signalk-raspberry-pi-1wire

**Comment by mgrouch** [comment]
And the third way (probably the best) is via SensEsp (requires ESP32 controller)
https://github.com/SignalK/SensESP

**Reply by mgrouch** [reply]
And if you have ESP32 then you can use either SensEsp (with SignalK) or EspHome (with HomeAssistant)

https://esphome.io/

**Comment by mgrouch** [comment]
Methods above will generate reading as SignalK messages. So you can add the to KIP dashboard.

There is also way to add them to HomeAssistant. You have it on BBN OS you need just activate it.

https://www.home-assistant.io/integrations/onewire/

For dashboards there are also options Grafana, SignalK KIP, NodeRed Dashboard, HomeAssistant

**Reply by ejbones** [reply]
Much appreciated.  What would be the benefit for using Homeassistant vs a just a signalk dashboard?

Mainly I'm just looking for gauges and alarms to start
